jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpCookie.java

```diff
@@ -72,313 +77,287 @@ public String getAttributeValue()
-    }
-
-    /**
-     * Names of well-known Attributes that are parsed by the constructors and shouldn't be present in the stored {@link #getAttributes()} Map.
-     */
-    private static final List<String> PARSED_ATTRIBUTE_NAMES = List.of("Domain", "Path", "Max-Age", "HttpOnly", "Secure", "Comment");
-
-    private final String _name;
-    private final String _value;
-    private final String _comment;
-    private final String _domain;
-    private final long _maxAge;
-    private final String _path;
-    private final boolean _secure;
-    private final int _version;
-    private final boolean _httpOnly;
-    private final long _expiration;
-    private final Map<String, String> _attributes;
-
-    /**
-     * Create new HttpCookie from specific values.
-     *
-     * @param name the name of the cookie
-     * @param value the value of the cookie
-     */
-    public HttpCookie(String name, String value)
-    {
-        this(name, value, -1);
-    }
-
-    /**
-     * Create new HttpCookie from specific values.
-     *
-     * @param name the name of the cookie
-     * @param value the value of the cookie
-     * @param domain the {@code Domain} value used for Domain-Matching rules on the cookie
-     * @param path the {@code Path} value to use for Path-Matching rules on the cookie
-     */
-    public HttpCookie(String name, String value, String domain, String path)
-    {
-        this(name, value, domain, path, -1, false, false);
-    }
-
-    /**
-     * Create new HttpCookie from specific values.
-     *
-     * @param name the name of the cookie
-     * @param value the value of the cookie
-     * @param maxAge the {@code Max-Age} attribute value (in seconds) for the cookie
-     */
-    public HttpCookie(String name, String value, long maxAge)
-    {
-        this(name, value, null, null, maxAge, false, false);
-    }
 
-    /**
-     * Create new HttpCookie from specific values.
-     *
-     * @param name the name of the cookie
-     * @param value the value of the cookie
-     * @param domain the {@code Domain} value used for Domain-Matching rules on the cookie
-     * @param path the {@code Path} value to use for Path-Matching rules on the cookie
-     * @param maxAge the {@code Max-Age} attribute value (in seconds) for the cookie
-     * @param httpOnly the {@code HttpOnly} attribute of the cookie
-     * @param secure the {@code Secure} attribute of the cookie
-     */
-    public HttpCookie(String name, String value, String domain, String path, long maxAge, boolean httpOnly, boolean secure)
-    {
-        this(name, value, domain, path, maxAge, httpOnly, secure, null, 0);
+        public static SameSite from(String sameSite)
+        {
+            if (sameSite == null)
+                return null;
+            return switch (StringUtil.asciiToLowerCase(sameSite))
+            {
+                case "lax" -> SameSite.LAX;
+                case "strict" -> SameSite.STRICT;
+                case "none" -> SameSite.NONE;
```

Comment:
I can imagine that the SameSite stuff is used very often in production, we should probably use one of our fancy Trie implementations for this step, no?